bundles/org.openhab.binding.surepetcare/src/main/java/org/openhab/binding/surepetcare/internal/dto/SurePetcareDeviceCurfewList.java

```diff
@@ -0,0 +1,89 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.surepetcare.internal.dto;
+
+import java.util.ArrayList;
+
+import org.openhab.binding.surepetcare.internal.SurePetcareConstants;
+
+/**
+ * The {@link SurePetcareDeviceCurfewList} class is used to serialize a list of curfew parameters.
+ *
+ * @author Rene Scherer - Initial contribution
+ */
+public class SurePetcareDeviceCurfewList extends ArrayList<SurePetcareDeviceCurfew> {
+
+    private static final long serialVersionUID = -6947992959305282143L;
+
+    /**
+     * Return the list element with the given index. If the list if too short, it will grow automatically to the given
+     * index.
+     *
+     * @return element with given index
+     */
+    @Override
+    public SurePetcareDeviceCurfew get(int index) {
+        if (index >= size()) {
```

Comment:
the flap can have up to 4 curfew times... the API then returns the times for each.

Reply:
But why does it need to grow when you try to fetch something. If you don't want to handle null it might make more sense to return an empty object instead of modifying the list. It might be unexpected that get actually modifies the list.

But given the time you have available, it might make more sense to leave this as is, given that it works now, and focus on the error handling / thing-status updating. Because this issue is about polishing they other or more important.